hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationFuture.java

```diff
@@ -63,19 +63,6 @@ protected TimeoutException newTimeoutException(long timeout, TimeUnit unit) {
     }
 
     @Override
-    protected Throwable unwrap(Throwable throwable) {
-        return throwable;
-    }
-
-    @Override
-    protected Object resolve(Object value) {
```

Comment:
This is the good part from the client: wrapping the remote exception in a generic exception. This is how it should be done in member/member in hz4.